packages/node-core/src/indexer/fetch.service.ts

```diff
@@ -313,14 +318,22 @@ export abstract class BaseFetchService<
 
       const enqueuingBlocks =
         handlers.length && this.getModulos().length === handlers.length
-          ? this.getEnqueuedModuloBlocks(startBlockHeight)
+          ? this.getEnqueuedModuloBlocks(startBlockHeight, latestHeight)
           : range(startBlockHeight, endHeight + 1);
 
       await this.enqueueBlocks(enqueuingBlocks);
     }
   }
 
   private async enqueueBlocks(enqueuingBlocks: number[]): Promise<void> {
+    // We check enqueuingBlocks length rather than cleanedBatchBlocks
+    // Because when bypass blocks, cleanedBatchBlocks can be [], but latestBufferHeight could be valid to update metadata
+    // See comments in blockDispatcher.enqueueBlocks method
+    if (!enqueuingBlocks.length) {
+      logger.info(`No blocks to enqueue at the moment.`);
+      await delay(10);
```

Comment:
I think we should remove this delay here, we should only have one place to throttle and this is a really large delay. If a chain has a block time of a couple of seconds this will add a big delay